billing-service/src/main/java/org/egov/demand/service/BillServicev2.java

```diff
@@ -208,7 +215,7 @@ public BillResponseV2 fetchBill(GenerateBillCriteria billCriteria, RequestInfoWr
 			billCriteria.getConsumerCode().retainAll(cosnumerCodesToBeExpired);
 			billCriteria.getConsumerCode().addAll(cosnumerCodesNotFoundInBill);
 			updateDemandsForexpiredBillDetails(billCriteria.getBusinessService(), billCriteria.getConsumerCode(), billCriteria.getTenantId(), requestInfoWrapper);
-			billRepository.updateBillStatus(cosnumerCodesToBeExpired, BillStatus.EXPIRED);
+			billRepository.updateBillStatus(cosnumerCodesToBeExpired,billCriteria.getBusinessService(), BillStatus.EXPIRED);
```

Comment:
was this tested? what about the scenario where there is no business service in criteria?

Reply:
Yes its tested. Business Service is not null.